src/provider/systemdmanager/systemdunit.rs

```diff
@@ -64,39 +127,79 @@ impl SystemDUnit {
 
         unit.name = format!("{}{}", name_prefix, trimmed_name);
 
-        unit.add_property(SECTION_UNIT, "Description", &unit.name.clone());
+        unit.add_property(Section::Unit, "Description", &unit.name.clone());
 
         unit.add_property(
-            SECTION_SERVICE,
+            Section::Service,
             "ExecStart",
-            &SystemDUnit::get_command(container, pod_state)?,
+            &SystemDUnit::get_command(container, template_data, package_root)?,
         );
 
-        let env_vars = SystemDUnit::get_environment(container, pod_state)?;
-
-        for (name, value) in env_vars {
-            unit.add_env_var(&name, &value);
+        let env_vars = SystemDUnit::get_environment(container, service_name, template_data)?;
+        if !env_vars.is_empty() {
+            let mut assignments = env_vars
+                .iter()
```

Comment:
Do we maybe want to sort these by key? Currently, if the ordering of these changes in the pod, the unit file would also change and trigger an inadvertend service restart, even though nothing substantially changed.

Arguably operators should take care of this, but that way we need to ensure this in every operator, this code is the last instance that sees these values before creating the unit file, so potentially a useful place to enforce deterministic behavior?

Reply:
The assignments are sorted by their keys, see line 144.

Reply:
true ... that is very weird, I added a test case with multiple env vars and switched them earlier, before writing my comment and swear that I saw the test case fail after switching them..
Anyway, comment withdrawn :)